src/core/chunked_stream.js

```diff
@@ -267,24 +242,483 @@ var ChunkedStream = (function ChunkedStreamClosure() {
+    };
+  };
+
+  ChunkedStreamContinuous.prototype.onReceiveProgressiveData = function
+      ChunkedStreamContinuous_onReceiveProgressiveDataContinuous(data) {
+
+    var chunkSize = this.chunkSize;
+    var position = this.progressiveDataLength;
+    var beginChunk = Math.floor(position / this.chunkSize);
+
+    this.buffer.buffer.set(new Uint8Array(data), position);
+    position += data.byteLength;
+    this.progressiveDataLength = position;
+    var endChunk = position >= this.end ? this.numChunks :
+        Math.floor(position / this.chunkSize);
+    var chunkEnd = chunkSize * beginChunk + data.byteLength;
+
+    for (var curChunk = beginChunk; curChunk < endChunk; ++curChunk) {
+      if (!(curChunk in this.loadedChunks)) {
+        this.loadedChunks[curChunk] = {
+          chunkOffset: chunkSize * curChunk,
+          start: 0,
+          startOffset: 0,
+          end: chunkEnd,
+          data: this.buffer.buffer
+        };
+        ++this.numChunksLoaded;
+      }
+    }
+    // Merge chunks to the left.
+    curChunk = beginChunk - 1;
+    while (curChunk >= 0 && this.loadedChunks[curChunk]) {
+      this.loadedChunks[curChunk--].end = chunkEnd;
+    }
+
+    if (this.numChunksLoaded === this.numChunks) {
+      // stream is fully loaded, switch to fast getByte
+      this.buffer.start = 0;
+      this.buffer.startOffset = 0;
+      this.buffer.end = this.buffer.buffer.byteLength;
+      this.prepareBuffer = this.prepareBufferNop;
+      this.getByte = this.getByteFastContinuous(this.buffer.buffer, this.end);
+    }
+  };
+
+  ChunkedStreamContinuous.prototype.prepareBuffer = function
+      ChunkedStreamContinuous_prepareBufferContinuous(begin, end) {
+    if (!end) {
+      return;
+    }
+
+    // Checks if current buffer matches new [begin, end) parameters.
+    if (this.buffer.start <= begin && end <= this.buffer.end ||
+        end <= this.progressiveDataLength) {
+      return;
+    }
+
+    // Checks if there is initial block
+    if (this.initialChunk && this.initialChunk.start <= begin &&
+        end <= this.initialChunk.end) {
+      this.buffer = this.initialChunk;
+      return;
+    }
+
+    var chunkSize = this.chunkSize;
+    var beginChunk = Math.floor(begin / chunkSize);
+    var endChunk = Math.floor((end - 1) / chunkSize) + 1;
+    // Check if there are missing chunks.
+    for (var chunk = beginChunk; chunk < endChunk; ++chunk) {
+      if (!this.loadedChunks[chunk]) {
+        throw new MissingDataException(begin, end);
+      }
+    }
+
+    this.buffer.startOffset = 0;
+    this.buffer.start = begin;
+    this.buffer.end = end;
+  };
+
+  ChunkedStreamContinuous.prototype.onReceiveData = function
+      ChunkedStreamContinuous_onReceiveDataContinuous(begin, chunk) {
+    var chunkSize = this.chunkSize;
+    var data = new Uint8Array(chunk);
+
+    if (begin === 0 && data.byteLength % chunkSize !== 0) {
+      this.initialChunk = {
+        start:  0,
+        startOffset: 0,
+        end:    0,
```

Comment:
Please make sure that all these occurrences (e.g., where the `:` is followed by more than one space) in the PR are changed to use exactly one space.